transitive_maven_jar/transitive_maven_jar.bzl

```diff
@@ -0,0 +1,55 @@
+
+DEPS = ["mvn"]
+MAX_TIMEOUT = 0x7FFFFFFF
+
+BINARY_URL = "https://github.com/petroseskinder/migration-tooling/releases/download/test/generate_workspace_deploy.jar"
+
+def _check_dependencies(rctx):
+	for dep in DEPS:
+		if rctx.which(dep) == None:
+			fail("%s requires %s as a dependency. Please check your PATH." % (rctx.name, dep))
+
+def _validate_coordinate_length(coordinate):
+	parts = coordinate.split(":")
+	return len(parts) >= 2 and len(parts) <= 5
+
+def _validate_coordinates(rctx):
+	coordinates = rctx.attr.artifacts
+	for coord in coordinates:
+		if _validate_coordinate_length(coord) == False:
+			fail("Invalid coordinate %s. Generally formatted as \"group:artifact:version\"" % coord)
+	return True
+
+def _create_arguments(rctx):
+	#TODO(petros): pass output directory as an attribute.
+	arguments = ['-a ' + artifact for artifact in rctx.attr.artifacts]
+	return ' '.join(arguments)
+
+def _execute(rctx, command_string):
+	return rctx.execute(["bash", "-c", command_string], timeout=rctx.attr.timeout)
+
+def _transitive_maven_jar_impl(rctx):
+	_check_dependencies(rctx)
+	_validate_coordinates(rctx)
+	arguments = _create_arguments(rctx)
+
+	# obtain the generate workspace binary
+	jar_path = rctx.path('generate_workspace_deploy.jar')
+	rctx.download(BINARY_URL, output=jar_path)
```

Comment:
You don't need to download this twice, use the one that was previously declared in the WORKSPACE file.